app/components/trigger-custom-build.js

```diff
@@ -0,0 +1,98 @@
+import Ember from 'ember';
+import { task } from 'ember-concurrency';
+import YAML from 'npm:yamljs';
+
+const { service } = Ember.inject;
+
+export default Ember.Component.extend({
+  ajax: service(),
+  flashes: service(),
+  router: service(),
+
+  classNames: ['trigger-build-modal'],
+  triggerBuildBranch: '',
+  triggerBuildMessage: '',
+  triggerBuildConfig: '',
+  triggering: false,
+
+  branches: Ember.computed.filterBy('repo.branches', 'exists_on_github', true),
+
+  didReceiveAttrs() {
+    this._super(...arguments);
+    this.set('triggerBuildBranch', this.get('repo.defaultBranch.name'));
+  },
+
+  sendTriggerRequest: task(function* () {
+    let config = YAML.parse(this.get('triggerBuildConfig'));
+
+    let runInterval = 0;
+    if (!Ember.testing) {
+      runInterval = 3000;
+    }
+
+    let body = {
+      request: {
+        branch: this.get('triggerBuildBranch'),
+        config: config
+      }
+    };
+
+    if (! Ember.isEmpty(this.get('triggerBuildMessage'))) {
+      body.request.message = this.get('triggerBuildMessage');
+    }
+
+    try {
+      this.get('ajax').postV3(`/repo/${this.get('repo.id')}/requests`, body)
+        .then((data) => {
+          let reqId = data.request.id;
+
+          Ember.run.later(this, function () {
```

Comment:
@backspace I tried splitting up the task into two but I still have trouble getting the yield and return right so that the task only returns when the 2nd task finishes. I think the Ember.run might be a problem?